ast-visual-studio-extension/CxExtension/CxWindowPackage.cs

```diff
@@ -66,7 +74,66 @@ protected override async Task InitializeAsync(CancellationToken cancellationToke
                 Console.WriteLine(ex.ToString());
             }
         }
+        private string getLogFilePath()
+        {
+            string dirName = Debugger.IsAttached ? "ast-visual-studio-extension-debug" : "ast-visual-studio-extension";
+            var logDirectory = Path.Combine(Path.GetTempPath(), dirName, "Logs");
+
+            if (!Directory.Exists(logDirectory))
+            {
+                Directory.CreateDirectory(logDirectory);
+            }
+
+            return Path.Combine(logDirectory, "ast-extension.log");
+        }
 
+        private string getLog4netConfigPath()
+        {
+            if (Debugger.IsAttached)
+            {
+                return Path.Combine(Environment.CurrentDirectory, "log4net.config");
+            }
+
+            string assemblyLocation = Assembly.GetExecutingAssembly().Location;
+            string baseDirectory = Path.GetDirectoryName(assemblyLocation);
+
+            string[] possiblePaths = new string[]
+            {
+                    Path.Combine(baseDirectory, "log4net.config"), // Installed extension location
+                    Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "log4net.config"), // Base directory
+            };
+            return possiblePaths.FirstOrDefault(File.Exists);
+        }
+
+        private void ConfigureLog4net()
+        {
+            try
+            {
+                string logFilePath = getLogFilePath();
+                GlobalContext.Properties["CxLogFileName"] = logFilePath;
+
+                string log4netConfigPath = getLog4netConfigPath();
+
+                if (log4netConfigPath == null)
+                {
+                    throw new FileNotFoundException("log4net.config not found in expected locations.");
+                }
+                var logRepository = LogManager.GetRepository();
+                XmlConfigurator.Configure(logRepository, new FileInfo(log4netConfigPath));
+
+                var appender = ((Hierarchy)logRepository).Root.Appenders[0];
+                if (appender is RollingFileAppender fileAppender && fileAppender.File != logFilePath)
+                {
+                    fileAppender.File = logFilePath;
+                    fileAppender.ActivateOptions();
+                }
+            }
+            catch (Exception ex)
+            {
+                var fallbackLogger = LogManager.GetLogger(typeof(CxWindowPackage));
+                fallbackLogger?.Error("Error during log4net configuration", ex);
```

Comment:
do we print this error to the console (in this case we don't have log file...)

Reply:
The fallbackLogger returns the configuration to its state before my config was applied. This means that if we don't have a log file configured, we fall back to the default logging behavior.